source/common/upstream/eds.cc

```diff
@@ -22,16 +24,21 @@ EdsClusterImpl::EdsClusterImpl(const Json::Object& config, Runtime::Loader& runt
 
 void EdsClusterImpl::initialize() { subscription_->start({cluster_name_}, *this); }
 
-bool EdsClusterImpl::onConfigUpdate(const ResourceVector& resources) {
+void EdsClusterImpl::onConfigUpdate(const ResourceVector& resources) {
   std::vector<HostSharedPtr> new_hosts;
   if (resources.size() != 1) {
-    ENVOY_LOG(warn, "Unexpected EDS resource length: {}", resources.size());
-    return false;
+    // We need to allow server startup to continue, even if we have a bad
+    // config.
+    runInitializeCallbackIfAny();
```

Comment:
It's a bit awkward at the level of https://github.com/lyft/envoy/blob/master/include/envoy/config/subscription.h to see why you would want this. How about I wrap the body of this function in another function which can catch the exception and then do the `runInitializeCallbackIfAny()` and then reraise?